src/Agent.Worker/ResourceMetricsManager.cs

```diff
@@ -38,78 +47,329 @@ public void Setup(IExecutionContext context)
+                    processStartInfo.Arguments = "-m";
+                    processStartInfo.RedirectStandardOutput = true;
+
+                    using (var process = Process.Start(processStartInfo))
+                    {
+                        processStartInfoOutput = process.StandardOutput.ReadToEnd();
+                    }
+
+                    var processStartInfoOutputString = processStartInfoOutput.Split("\n");
+                    var memoryInfoString = processStartInfoOutputString[1].Split(" ", StringSplitOptions.RemoveEmptyEntries);
+
+                    if (memoryInfoString.Length != 7)
+                    {
+                        throw new MemoryMonitoringUtilityIsNotAvaliableException("Utility has non-default output");
+                    }
+
+                    memoryInfo.TotalMemoryMB = Int32.Parse(memoryInfoString[1]);
+                    memoryInfo.UsedMemoryMB = Int32.Parse(memoryInfoString[2]);
+                }
+                catch (Win32Exception e)
+                {
+                    throw new MemoryMonitoringUtilityIsNotAvaliableException(e.Message);
+                }
+            }
+
+            if (PlatformUtil.RunningOnMacOS)
+            {
+                processStartInfo.FileName = "vm_stat";
+                processStartInfo.RedirectStandardOutput = true;
+
+                using (var process = Process.Start(processStartInfo))
+                {
+                    processStartInfoOutput = process.StandardOutput.ReadToEnd();
+                }
+
+                var processStartInfoOutputString = processStartInfoOutput.Split("\n");
+
+                var pageSize = Int32.Parse(processStartInfoOutputString[0].Split(" ", StringSplitOptions.RemoveEmptyEntries)[7]);
+
+                var pagesFree = Int64.Parse(processStartInfoOutputString[1].Split(" ", StringSplitOptions.RemoveEmptyEntries)[2].Trim('.'));
+                var pagesActive = Int64.Parse(processStartInfoOutputString[2].Split(" ", StringSplitOptions.RemoveEmptyEntries)[2].Trim('.'));
+                var pagesInactive = Int64.Parse(processStartInfoOutputString[3].Split(" ", StringSplitOptions.RemoveEmptyEntries)[2].Trim('.'));
+                var pagesSpeculative = Int64.Parse(processStartInfoOutputString[4].Split(" ", StringSplitOptions.RemoveEmptyEntries)[2].Trim('.'));
+                var pagesWiredDown = Int64.Parse(processStartInfoOutputString[6].Split(" ", StringSplitOptions.RemoveEmptyEntries)[3].Trim('.'));
+                var pagesOccupied = Int64.Parse(processStartInfoOutputString[16].Split(" ", StringSplitOptions.RemoveEmptyEntries)[4].Trim('.'));
+
+                var freeMemory = (pagesFree + pagesInactive) * pageSize;
+                var usedMemory = (pagesActive + pagesSpeculative + pagesWiredDown + pagesOccupied) * pageSize;
+
+                memoryInfo.TotalMemoryMB = (int)((freeMemory + usedMemory) / 1048576);
+                memoryInfo.UsedMemoryMB = (int)(usedMemory / 1048576);
```

Comment:
all of the  int32 references to memory will overflow if >4 gigs.  Use ulong instead.

Reply:
Hmm... this actually was tested and worked correctly on the machine with 16GB, but indeed it may overflow on bigger values, thanks for noticing! I'll fix this.
Also, I think signed long will be enough, since it has maximum value of 9,223,372,036,854,775,807 and for Mac machines with let's say 1.5TB of RAM with page size of 16k the maximum value of `freeMemory + usedMemory` before the division will be 1,610,612,736,000 which give us enough free range.